File: tournaments/longestString/longestString.cpp
std::string longestString(std::vector<std::string> inputArray) {

  std::string answer = inputArray[0];
  for (int i = 1; i < inputArray.size(); i++) {
    if (inputArray[i].size() > answer.size()) {
      answer = inputArray[i];
    }
  }
  return answer;
}

File: tournaments/longestString/longestString.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub longestString {
    my ($inputArray) = @_;

    my $result = "";

    foreach my $item (@$inputArray) {
        if (length $result < length $item) {
            $result = $item;
        }
    }

    return $result;
}
